connectors/src/connectors/google_drive/temporal/workflows.ts

```diff
@@ -177,6 +177,13 @@ export async function googleDriveIncrementalSync(
         startSyncTs,
         nextPageToken
       );
+
+      if (workflowInfo().historyLength > 4000) {
```

Comment:
But this will just restart the workflow losing the whole state no ? it will never complete like this

Reply:
Updated.